clang-tools-extra/clangd/XRefs.cpp

```diff
@@ -430,6 +430,18 @@ locateASTReferent(SourceLocation CurLoc, const syntax::Token *TouchedIdentifier,
         continue;
       }
     }
+    // Special case: an Objective-C method can override a parent class or
+    // protocol.
```

Comment:
what about the interaction in the other direction? i.e. we're on the protocol declaration and try to do a go-to-definition.

moreover this is going to make go-to-defn on references to a symbol quite chatty (now you'll have both the definition in implementation and definition in protocol as an alternative). I think we want to limit this to only when user triggered the interaction on the definition of the method and not references to it (similar to c++ code path above).

Reply:
I was thinking for that users can use go to implementation to find them. 

And that's a good point - didn't think about that, updated.

Reply:
> I was thinking for that users can use go to implementation to find them.

i think using go-to-definition for such interactions is a more common workflow (also go-to-impl is not necessarily implemented by all editors). so I think there's merit in making sure that also works (though I won't block this patch on it, as it's not a regression. but I think it warrants at least a `FIXME`)